static/js/lib/ckeditor/plugins/Markdown.ts

```diff
@@ -132,6 +132,25 @@ export default class Markdown extends MarkdownConfigPlugin {
     )
   }
 
+  /**
+   * The @ckeditor/ckeditor5-markdown-gfm package uses a single turndownService
+   * instance for all instances of GFMDataProcessor. So sad.
+   *
+   * This is frustrating because we want different editors to support different
+   * markdown syntaxes (e.g., "minimal" editors do not support table editing.)
+   *
+   * So we'll emulate separate instances by swapping out the rulesset on the
+   * single instance.
+   */
+  turndown = (html: string) => {
+    try {
+      turndownService.rules.array = this.turndownRules
+      return turndownService.turndown(html)
+    } finally {
+      turndownService.rules.array = BASE_TURNDOWN_RULES
+    }
+  }
+
```

Comment:
It's not entirely clear to me that overriding `rules.array` in this manner is part of Turndown's public API. But we're already doing it https://github.com/mitodl/ocw-studio/blob/8d992c7f28d051e8235636a57a41c4f096db09ac/static/js/lib/ckeditor/turndown.ts#L26

Reply:
The functions in Turndown for adding rules manipulate the `rules` array, so I don't see any issues with directly manipulating it ourselves.

Reply:
Yeah, but it always makes me nervous... Does Turndown consider this an implementation detail that might change in a minor release? Whereas if they change how `addRule` behaves, that's clearly a breaking change for their public API.

But... if it idid break in a minor release, our tests would catch it.